modules/client/src/client-common/utils.ts

```diff
@@ -121,3 +126,50 @@ export function votingModeFromContracts(votingMode: number): VotingMode {
       throw new InvalidVotingModeError();
   }
 }
+
+export function isFailingProposal(actions: DaoAction[]): boolean {
+  const functionNames: string[] = [];
+  // store the function names of the actions
+  for (const action of actions) {
+    try {
+      const fragment = getFunctionFragment(
+        action.data,
+        FAILING_PROPOSAL_AVAILABLE_FUNCTION_SIGNATURES,
+      );
+      console.log(fragment);
+      functionNames.push(
+        fragment.name,
+      );
+    } catch (e) {
+      console.log(e);
+    }
+  }
+  for (const [i, functionName] of functionNames.entries()) {
```

Comment:
What would happen if updateSettings was unchanged and an address was removed/added (while settings being still valid)?

Are we reproducing the exact same logic that would fail on the plugins side? Shouldn't we somehow check the current minApprovals and the new one?

Reply:
And what if there was some other setting (non-wallet related) that was changed? I still believe we should inspect a bit deeper what's actually changing, but I may be missing context

Reply:
I mean there is not much more to do that can break a proposal, the whole reasoning behind this was avoiding proposals that increase or reduce the number of adddresses without changing the settings leading to a bricked dao ora proposal creation error. We can go deeper and decode the data and check if the settings would make sense instead of just checking the order and name functions

Reply:
As long as we are sure that we are not giving any false positives, then let's go with it and improve over time